src/libs/actions/Welcome.ts

```diff
@@ -2,139 +2,84 @@ import type {OnyxCollection} from 'react-native-onyx';
 import Onyx from 'react-native-onyx';
 import type {OnboardingPurposeType} from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
+import type Onboarding from '@src/types/onyx/Onboarding';
 import type OnyxPolicy from '@src/types/onyx/Policy';
 import type {EmptyObject} from '@src/types/utils/EmptyObject';
 
-let hasSelectedPurpose: boolean | undefined;
-let hasProvidedPersonalDetails: boolean | undefined;
-let isFirstTimeNewExpensifyUser: boolean | undefined;
+let onboarding: Onboarding | [] | undefined;
 let hasDismissedModal: boolean | undefined;
 let isLoadingReportData = true;
 
-type DetermineOnboardingStatusProps = {
-    onAble?: () => void;
-    onNotAble?: () => void;
-};
-
 type HasCompletedOnboardingFlowProps = {
     onCompleted?: () => void;
     onNotCompleted?: () => void;
 };
 
 let resolveIsReadyPromise: (value?: Promise<void>) => void | undefined;
-const isServerDataReadyPromise = new Promise<void>((resolve) => {
+let isServerDataReadyPromise = new Promise<void>((resolve) => {
     resolveIsReadyPromise = resolve;
 });
 
 let resolveOnboardingFlowStatus: (value?: Promise<void>) => void | undefined;
-const isOnboardingFlowStatusKnownPromise = new Promise<void>((resolve) => {
+let isOnboardingFlowStatusKnownPromise = new Promise<void>((resolve) => {
     resolveOnboardingFlowStatus = resolve;
 });
 
 function onServerDataReady(): Promise<void> {
     return isServerDataReadyPromise;
 }
 
-/**
- * Checks if Onyx keys required to determine the
- * onboarding flow status have been loaded (namely,
- * are not undefined).
- */
-function isAbleToDetermineOnboardingStatus({onAble, onNotAble}: DetermineOnboardingStatusProps) {
-    const hasRequiredOnyxKeysBeenLoaded = [hasProvidedPersonalDetails, hasSelectedPurpose].every((value) => value !== undefined);
-
-    if (hasRequiredOnyxKeysBeenLoaded) {
-        onAble?.();
-    } else {
-        onNotAble?.();
-    }
-}
-
-/**
- * A promise returning the onboarding flow status.
- * Returns true if user has completed the onboarding
- * flow, false otherwise.
- */
 function isOnboardingFlowCompleted({onCompleted, onNotCompleted}: HasCompletedOnboardingFlowProps) {
     isOnboardingFlowStatusKnownPromise.then(() => {
-        // Remove once Stage 1 Onboarding Flow is ready
-        if (!isFirstTimeNewExpensifyUser) {
+        if (Array.isArray(onboarding) || onboarding?.hasCompletedGuidedSetupFlow === undefined) {
```

Comment:
NAB: Can you please add a comment to some follow up PR to explain that the `onboarding` is an empty array when the user hasnt completed the flow yet

Reply:
Will add in some follow up